eng/pipelines/publish-dev-release-go.yml

```
@@ -15,5 +15,9 @@ steps:
       cd packages/autorest.go
       npm version --no-git-tag-version $currentVersion
       npm pack;
+      npm install -g autorest-go-$currentVersion.tgz
+      if (!$?) {
```

@benpb does this ealy exit look correct to you?

---

@jhendrixMSFT this works but I prefer:

```suggestion
      if ($LASTEXITCODE) {
        exit $LASTEXITCODE
      }
```

---

I also like to add this to the top of scripts sometimes:

```
$ErrorActionPreference = 'Stop'  # on error halt execution
$PSNativeCommandUseErrorActionPreference = $true  # on non-powershell binary failure inherit error action preference
```

However this doesn't work well with some cli tools that like to add to the stderr stream, as powershell treats those as failure. So then you have to either do `2>&1` or just handle LASTEXITCODE.